src/main/java/KBot.java

```diff
@@ -0,0 +1,211 @@
+                if (!argument.contains("/from ") || !argument.contains("/to ")) {
+                    throw new KBotException("The event description or timing is missing.");
+                } else {
+                    addEventTask(argument);
+                }
+                break;
+            case "delete":
+                try {
+                    deleteTask(Integer.parseInt(argument));
+                } catch (NumberFormatException e) {
+                    throw new KBotException("Invalid task number to delete.");
+                }
+                break;
+            default:
+                throw KBotException.unknownCommand();
+        }
+        return true;
+    }
+
+
+    // Utility Methods
+
+    private void greetUser() {
+        printSeparator();
+        System.out.println("Hello! I'm KBot");
+        System.out.println("What can I do for you?");
+        printSeparator();
+    }
+
+    private void printSeparator() {
+        System.out.println(SEPARATOR);
+    }
+
+    private String getUserInput() {
+        return scanner.nextLine();
+    }
+
+    private void addTodoTask(String description) {
+        Task task = new Todo(description);
+        tasks.add(task);
+        printTaskAddedMessage(task);
+    }
+
+    private void addDeadlineTask(String input) throws KBotException {
+        String[] parts = input.split(" /by ");
+        if (parts.length < 2) {
+            throw new KBotException("Invalid format for deadline. Use: deadline <description> /by <time>");
+        }
+        Task task = new Deadline(parts[0], parts[1]);
+        tasks.add(task);
+        printTaskAddedMessage(task);
+    }
+
+    private void addEventTask(String input) throws KBotException {
+        String[] parts = input.split(" /from | /to ");
+        if (parts.length < 3) {
+            throw new KBotException("Invalid format for event. Use: event <description> /from <start> /to <end>");
+        }
+        Task task = new Event(parts[0], parts[1], parts[2]);
+        tasks.add(task);
+        printTaskAddedMessage(task);
+    }
+
+    private void listTasks() {
+        printSeparator();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + "." + tasks.get(i));
+        }
+        printSeparator();
+    }
+
+    private void markTaskAsDone(int index) throws KBotException {
+        if (index >= tasks.size() || index < 0) {
+            throw new KBotException("Invalid task number.");
+        }
+        tasks.get(index).markAsDone();
+        printMarkDoneMessage(tasks.get(index));
+    }
+
+    private void markTaskAsNotDone(int index) throws KBotException {
+        if (index >= tasks.size() || index < 0) {
+            throw new KBotException("Invalid task number.");
+        }
+        tasks.get(index).markAsNotDone();
+        printMarkNotDoneMessage(tasks.get(index));
+    }
+
+    private void showError(String errorMessage) {
```

Comment:
Perhaps be consistent in the naming of the methods that involves printing?
```suggestion
    private void printError(String errorMessage) {
```